### PR TITLE
[16.0][ADD]Base dns infrastructure

### PR DESCRIPTION
**Pull request by flotho**
trying a standard port with oca-port

### CODE REVIEW

**Comment by flotho** [comment]
Hi,
I would like to submit for a first review this PR to check if I'm going in the right direction.
@lasley @noahzaozao @ruter-lyu @murtuzasaleh @max3903 @JoJoJoJoJoJoJo all comments would will be appreciated.

The next step will be to create connector modules, I will start with an OVH connector

**Comment by flotho** [comment]
ping @sbidoul there's an issue during the build and I'm not able to identify the origin.
Regards

**Comment by flotho** [comment]
here is the logs that I can't see what to to from my side : 
```
ERROR: Cannot install odoo<16.1dev and >=16.0a and odoo<9.0a and >=8.0a because these package versions have conflicting dependencies.

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
The conflict is caused by:
    The user requested odoo<16.1dev and >=16.0a
    The user requested odoo<9.0a and >=8.0a
```

**Comment by sbidoul** [comment]
Hi @flotho! That is because there is a [manifest with version 8.0](https://github.com/OCA/infrastructure/pull/18/files#diff-c2e0f92cd369f25bb042e530081acb0378824ebf349cc811553eb4ae075abd37R7) in this PR. Addons with such manifest automatically depend on Odoo 8 and pip cannot satisfy that in a 16.0 branch CI.

**Comment by flotho** [comment]
thanks @sbidoul , I'll check this

**Comment by flotho** [comment]
Hi @lasley @noahzaozao @ruter-lyu @murtuzasaleh @max3903 @JoJoJoJoJoJoJo 

I'm rebooting this repo in v16 with a simple module as a strating point.
Everything is green here
The next step will be to create connector modules, 
Please feel free to comment and reviw.

Thanks

**Comment by bosd** [comment]
Can you fix pre-commit?

**Comment by flotho** [comment]
> Can you fix pre-commit?

with pleasure,
Done !

**Comment by flotho** [comment]
hum @bosd , did I miss something on the pylint issue?
Not sur I can do something here!

**Comment by bosd** [comment]
> hum @bosd , did I miss something on the pylint issue?
> Not sur I can do something here!

Hmm, maybe this repo needs an update with `copier update`

**Comment by flotho** [comment]
> > hum @bosd , did I miss something on the pylint issue?
> > Not sur I can do something here!
> 
> Hmm, maybe this repo needs an update with `copier update`

Thanks for the proposal, sadly the command `copier update` failed. Do you have any tips/doc/recommandation on how to use copier in OCA projects ?

**Comment by flotho** [comment]
not sure to understand what's wrong in this pre commit

**Comment by bosd** [comment]
Ooh I think the copier update' of the repo has already been done recently.

Try rebasing your branch. That should fix it.

File: base_dns_infrastructure/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Base module for DNS infrastructure",
+    "version": "16.0.1.0.1",
+    "category": "Services",
+    "application": True,
+    "external_dependencies": {
+        "python": ["validator_collection"],
```
Comment (bosd) [inline base_dns_infrastructure/__manifest__.py:9]:
```suggestion
        "python": ["validator-collection"],
```

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# generated from manifests external_dependencies
+validator_collection
```
Comment (bosd) [inline requirements.txt:2]:
```suggestion
validator-collection
```

File: .pylintrc-mandatory
Comment (bosd) [inline .pylintrc-mandatory:1]: Why change any of the `.` dotfiles?
Maybe better to exclude the changes, for the CI.

File: base_dns_infrastructure/static/description/icon.png
Comment (bosd) [inline base_dns_infrastructure/static/description/icon.png:1]: You can replace (overwrite) this file, with the new icon you created.

File: base_dns_infrastructure/data/dns_record_type.xml [hunk 3]
```diff
@@ -0,0 +1,159 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="1">
+    <record model="ir.module.category" id="module_category_infrastructure_dns">
+        <field name="name">Infrastructure et DNS</field>
```
Comment (bosd) [inline base_dns_infrastructure/data/dns_record_type.xml:4]:
```suggestion
        <field name="name">Infrastructure and DNS</field>
```

**Review by bosd — CHANGES_REQUESTED** [review]
very quick code review. The suggested changes should make gh actions happy. :green_heart:

**Comment by flotho** [comment]
closed in favor of https://github.com/OCA/infrastructure/pull/20

**Pull request closed without merging**